transcom/mymove
# MB-7650 Service counseling edit allowances frontend

## Description

This PR creates a new form specifically for the service counseling to edit the allowances. Changes to the allowances form also affects the allowances form for the TXO portion.

> As an office user
> When I am on the service counselor move details page for a move
> Then I am able to make edits to allowances including the following fields:
> Branch, Rank, pro-gear, spouse pro-gear, RME, and OCIE

## Reviewer Notes

Is there anything you would like reviewers to give additional scrutiny?
- I made a new document wrapper specifically for the services counseling
- We're still using the same form `AllowancesDetailForm` for both TXO and Service counseling
- Authorized weight is only editable for the TXO roles

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
make db_dev_e2e_populate
make server_run
make office_client_run
```

Test
1. login as Service Counseling role
2. Navigate to `/counseling/moves/queue`
3. Click on any move with status `Needs counseling`
4. Click on the `Edit Allowances` button
5. Modify inputs and Save

Check out new changes in storybook
1. Run `make storybook`
2. Navigate to `http://localhost:6006/?path=/story/office-components-allowancesdetailform--basic`
3. Play with the params in the `Controls` section
4. `editableAuthorizedWeight` is to enable/disable authorized weight input
5. `header` is to show any string you type in

## Code Review Verification Steps


## References

* [Jira story](https://dp3.atlassian.net/browse/MB-7650) for this change

## Screenshots


https://user-images.githubusercontent.com/1522549/116590424-96b56180-a8d2-11eb-9b9a-58d120f62e02.mov
